### PR TITLE
Fix the build for NLL

### PR DESCRIPTION
**Pull request by SimonSapin**
Test with `RUSTFLAGS="-Zborrowck=mir -Ztwo-phase-borrows" cargo build`

https://internals.rust-lang.org/t/help-us-get-non-lexical-lifetimes-nll-over-the-finish-line/7807/7

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/21356)

### CODE REVIEW

**Review by emilio — APPROVED** [review]
Looks good :)

File: components/layout/table.rs [hunk 1]
```diff
@@ -448,9 +448,9 @@ impl Flow for TableFlow {
                         .sum::<f32>();
 
                     for &index in &constrained_column_inline_sizes_indices {
+                        let inline_size = self.column_computed_inline_sizes[index].size.0;
                         self.column_computed_inline_sizes[index].size +=
-                            remaining_inline_size.scale_by(
-                                self.column_computed_inline_sizes[index].size.0 as f32 / total_minimum_size);
```
[the remark this answers is not in the document]
Comment (KiChjang) [reply components/layout/table.rs]: This is possibly rust-lang/rust#47349, which I'm working on.

[the remark this answers is not in the document]
Comment (SimonSapin) [reply components/layout/table.rs]: Per https://github.com/rust-lang/rust/issues/47349#issuecomment-362295502 the new borrowck behavior is considered correct. That issue is only still open in order to improve the error message.

File: components/layout/table.rs [hunk 2]
```diff
@@ -531,7 +531,7 @@ impl Flow for TableFlow {
         self.block_flow.build_display_list_for_block(state, border_painting_mode);
 
         let iter = TableCellStyleIterator::new(&self);
-        for mut style in iter {
+        for style in iter {
```
[the remark this answers is not in the document]
Comment (SimonSapin) [reply components/layout/table.rs]: Yes, I suppose that they were false negatives for the warning under AST-borrowck.

**Comment by SimonSapin** [comment]
Oops I accidentally pushed this to master :/ Thanks @jdm for taking care of reverting. Don’t we have github settings to prevent this from happening, though?

**Comment by SimonSapin** [comment]
Per https://github.com/servo/servo/issues/21358 it looks like unrelated changes in `url` fixed some tests, so this PR will need to update expectations.

@bors-servo try

**Comment by jdm** [comment]
@bors-servo retry
* cleaned out the builder

**Comment by jdm** [comment]
@bors-servo retry
* #20734

**Comment by jdm** [comment]
This continues to be broken on every single android-x86 run.

**Comment by jdm** [comment]
@bors-servo retry
* #20734 #21409

**Comment by jdm** [comment]
@bors-servo r-
I keep forgetting about the android problem.

**Comment by SimonSapin** [comment]
@jdm what’s the android problem?

Maybe I’d find out by trying:

@bors-servo r=jdm

**Comment by jdm** [comment]
@SimonSapin http://servo-master.servo.org:54856/results/servo/21356 reinforces https://github.com/servo/servo/pull/21356#issuecomment-411902287.

**Comment by SimonSapin** [comment]
I’ve backported the NLL fix in the `url` crate as 1.6.1, to avoid bringing in (in this PR) unrelated changes.

There is now the `gstreamer` crate whose current versions fail to build with NLL. This is fixed in git upstream, but not yet on crates.io versions. I’ve asked for a backport: https://github.com/sdroege/gstreamer-rs/commit/dba110e8ad255f7327d9f70cd7565917a303df12

Still, the rest of this PR can be landed without waiting for that.

@bors-servo r=jdm

(Carrying previous r+ since the PR is now a subset of the changes since the previous review.)

**Comment by SimonSapin** [comment]
Same as before:

```
 5:03.86 ERROR Traceback (most recent call last):
  File "/home/servo/buildbot/slave/android-x86/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/executors/protocol.py", line 47, in setup
    self.connect()
  File "/home/servo/buildbot/slave/android-x86/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/executors/executorservodriver.py", line 97, in connect
    wait_for_service((self.host, self.port), timeout=self.init_timeout)
  File "/home/servo/buildbot/slave/android-x86/build/tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/webdriver_server.py", line 265, in wait_for_service
    raise socket.error("Service is unavailable: %s:%i" % addr)
error: Service is unavailable: 127.0.0.1:4444
```

So my assumption that this was due to changes in the URL crate (perhaps in the webdriver server implementation) was not correct. This issue will need more investigation. (getopts might be another lead to follow.)

**Comment by SimonSapin** [comment]
*Sigh*. `getopts` silently removed support for `--webdriver 4444` (instead of `--webdriver=4444`) in a micro version number increment.

Added `=` in a few places, r?

**Comment by SimonSapin** [comment]
WPT manifest updated.

@bors-servo r=jdm

**Pull request merged**